integrations/jina/tests/test_document_embedder.py

```diff
@@ -0,0 +1,239 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+import json
+from unittest.mock import patch
+
+import pytest
+import requests
+from haystack import Document
+
+from jina_haystack.document_embedder import JinaDocumentEmbedder
+
+
+def mock_session_post_response(*args, **kwargs):  # noqa: ARG001
+    inputs = kwargs["json"]["input"]
+    model = kwargs["json"]["model"]
+    mock_response = requests.Response()
+    mock_response.status_code = 200
+    data = [{"object": "embedding", "index": i, "embedding": [0.1, 0.2, 0.3]} for i in range(len(inputs))]
+    mock_response._content = json.dumps(
+        {"model": model, "object": "list", "usage": {"total_tokens": 4, "prompt_tokens": 4}, "data": data}
+    ).encode()
+
+    return mock_response
+
+
+class TestJinaDocumentEmbedder:
+    def test_init_default(self, monkeypatch):
+        monkeypatch.setenv("JINA_API_KEY", "fake-api-key")
+        embedder = JinaDocumentEmbedder()
+
+        assert embedder.model_name == "jina-embeddings-v2-base-en"
+        assert embedder.prefix == ""
+        assert embedder.suffix == ""
+        assert embedder.batch_size == 32
+        assert embedder.progress_bar is True
+        assert embedder.metadata_fields_to_embed == []
+        assert embedder.embedding_separator == "\n"
+
+    def test_init_with_parameters(self):
+        embedder = JinaDocumentEmbedder(
+            api_key="fake-api-key",
+            model_name="model",
+            prefix="prefix",
+            suffix="suffix",
+            batch_size=64,
+            progress_bar=False,
+            metadata_fields_to_embed=["test_field"],
+            embedding_separator=" | ",
+        )
+        assert embedder.model_name == "model"
+        assert embedder.prefix == "prefix"
+        assert embedder.suffix == "suffix"
+        assert embedder.batch_size == 64
+        assert embedder.progress_bar is False
+        assert embedder.metadata_fields_to_embed == ["test_field"]
+        assert embedder.embedding_separator == " | "
+
+    def test_init_fail_wo_api_key(self, monkeypatch):
+        monkeypatch.delenv("OPENAI_API_KEY", raising=False)
```

Comment:
```suggestion
```

This only makes sense for the OpenAI Embedders.

Reply:
Changed the key